internal/lldp/tx.go

```diff
@@ -0,0 +1,270 @@
+		})
+	}
+
+	// EndTLV will be added by the serializer, we don't need to add it here
+	return r
+}
+
+func (l *LLDP) setUpTx() error {
+	l.mu.Lock()
+	defer l.mu.Unlock()
+	// Check if already set up (double-check pattern to prevent duplicate setup)
+	if l.tPacketTx != nil {
+		return nil
+	}
+
+	logger := l.l.With().Str("interface", l.interfaceName).Logger()
+	tPacketTx, err := afPacketNewTPacket(l.interfaceName)
+	if err != nil {
+		return err
+	}
+	logger.Info().Msg("created TPacket instance for sending LLDP packets")
+
+	l.tPacketTx = tPacketTx
+
+	return nil
+}
+
+func (l *LLDP) sendTxPackets() error {
+	l.mu.RLock()
+	defer l.mu.RUnlock()
+
+	logger := l.l.With().Str("interface", l.interfaceName).Logger()
+	iface, err := net.InterfaceByName(l.interfaceName)
+	if err != nil {
+		return err
+	}
+
+	if l.tPacketTx == nil {
+		return fmt.Errorf("AFPacket not initialized")
+	}
+
+	// create payload
+	ethFrame := layers.Ethernet{
+		EthernetType: lldpEtherType,
+		SrcMAC:       iface.HardwareAddr,
+		DstMAC:       lldpDstMac,
+	}
+
+	lldpFrame := layers.LinkLayerDiscovery{
+		ChassisID: layers.LLDPChassisID{
+			Subtype: layers.LLDPChassisIDSubTypeMACAddr,
+			ID:      []byte(iface.HardwareAddr),
+		},
+		PortID: layers.LLDPPortID{
+			Subtype: layers.LLDPPortIDSubtypeIfaceName,
+			ID:      []byte(iface.Name),
+		},
+		TTL:    uint16(3600),
+		Values: l.toPayloadValues(),
+	}
+
+	buf := gopacket.NewSerializeBuffer()
+	if err := gopacket.SerializeLayers(buf, gopacket.SerializeOptions{
+		FixLengths:       true,
+		ComputeChecksums: true,
+	}, &ethFrame, &lldpFrame); err != nil {
+		l.l.Error().Err(err).Msg("unable to serialize packet")
+		return err
+	}
+
+	logger.Trace().Hex("packet", buf.Bytes()).Msg("sending LLDP packet")
+
+	// send packet
+	if err := l.tPacketTx.WritePacketData(buf.Bytes()); err != nil {
+		l.l.Error().Err(err).Msg("unable to send packet")
+		return err
+	}
+
+	return nil
+}
+
+const txInterval = 30 * time.Second // Standard LLDP transmission interval
+
+func (l *LLDP) doSendPeriodically(logger *zerolog.Logger, txCtx context.Context) {
+	l.mu.Lock()
+	l.txRunning = true
+	l.mu.Unlock()
+
+	defer func() {
+		l.mu.Lock()
+		l.txRunning = false
+		l.mu.Unlock()
+	}()
+
+	ticker := time.NewTicker(txInterval)
+	defer ticker.Stop()
+
+	// Send initial packet immediately
+	if err := l.sendTxPackets(); err != nil {
+		logger.Error().Err(err).Msg("error sending initial LLDP packet")
+	}
+
+	for {
+		select {
+		case <-ticker.C:
+			if err := l.sendTxPackets(); err != nil {
+				logger.Error().Err(err).Msg("error sending LLDP packet")
+			}
+		case <-txCtx.Done():
+			logger.Info().Msg("LLDP transmitter stopped")
+			return
+		}
+	}
+}
+
+func (l *LLDP) startTx() error {
+	l.mu.RLock()
+	running := l.txRunning
+	enabled := l.enableTx
+	cancel := l.txCancel
+	l.mu.RUnlock()
+
+	if running || !enabled {
+		return nil
+	}
+
+	if cancel != nil {
+		cancel()
+	}
+
+	l.txCtx, l.txCancel = context.WithCancel(context.Background())
```

Comment:
This code mutates the LLDP object, shouldn't it be under an `l.mu.Lock()` wit the corresponding `l.mu.Unlock() just before like 239 (the doSend call)?